source/header.c

```diff
@@ -474,6 +472,10 @@ static void init_aws_byte_buf_raw(struct aws_byte_buf *buf) {
 int aws_cryptosdk_hdr_write(
     const struct aws_cryptosdk_hdr *hdr, size_t *bytes_written, uint8_t *outbuf, size_t outlen) {
     AWS_PRECONDITION(aws_cryptosdk_hdr_is_valid(hdr));
+    AWS_PRECONDITION(hdr->iv.len <= 255);  // uint8_t max value
+    AWS_PRECONDITION(
+        outlen == 0 || AWS_MEM_IS_READABLE(outbuf, outlen), "Output buffer [outbuf] must be readable up to [outlen].");
```

Comment:
Nice documentation, but I think the code is self explanatory, so we can keep it without this string following the same pattern of the other conditions.

Reply:
Updated!